base_layer/p2p/src/domain_message.rs

```diff
@@ -42,4 +43,34 @@ impl<T> DomainMessage<T> {
     pub fn into_inner(self) -> T {
         self.inner
     }
+
+    /// Converts the wrapped value of a DomainMessage to another compatible type.
+    ///
+    /// Note:
+    /// The Rust compiler doesn't seem to be able to recognise that DomainMessage<T> != DomainMessage<U>, so a blanket
+    /// `From` implementation isn't possible at this time
+    pub fn convert<U>(self) -> DomainMessage<U>
```

Comment:
That's a bummer - suppose it's because T could be U and there's no language feature express that the compile should fail if it is - this is a good compromise :+1:

Reply:
Ah.. that could be why.

We need a `where T: From<U> + T is_not U` type syntax :)